core/src/main/java/com/klaytn/caver/tx/type/TxTypeFeeDelegate.java

```diff
@@ -16,9 +16,160 @@
+            List<RlpType> vrs = ((RlpList) signatureRlpType).getValues();
+            if (vrs.size() < 3) continue;
+            byte[] v = ((RlpString) vrs.get(0)).getBytes();
+            byte[] r = ((RlpString) vrs.get(1)).getBytes();
+            byte[] s = ((RlpString) vrs.get(2)).getBytes();
+            addFeePayerSignatureData(new KlaySignatureData(v, r, s));
+        }
+    }
+
+    /**
+     * add signature data
+     *
+     * @param values rlp encoded rawTransaction
+     * @param offset where sender's signature data begins
+     */
+    public void addSignatureData(List<RlpType> values, int offset) {
+        if (values.size() > offset) {
+            List<RlpType> senderSignatures = ((RlpList) (values.get(offset))).getValues();
+            addSenderSignatureData(senderSignatures);
+        }
+
+        if (values.size() > offset + 1) {
+            String feePayer = ((RlpString) values.get(offset + 1)).asString();
+            setFeePayer(feePayer);
+        }
+
+        if (values.size() > offset + 2) {
+            List<RlpType> feePayerSignatures = ((RlpList) (values.get(offset + 2))).getValues();
+            addFeePayerSignatureData(feePayerSignatures);
+        }
+    }
+
+    /**
+     * add signature data
+     *
+     * @param txType TxType holding a signature
+     */
+    public void addSignatureData(TxTypeFeeDelegate txType) {
+        addSenderSignatureData(txType.getSenderSignatureDataSet());
+        addFeePayerSignatureData(txType.getSenderSignatureDataSet());
+    }
+
+    /**
+     * rlp encoding for transaction hash(TxHash)
+     *
+     * @param credentials credential info of a signer
+     * @param chainId     chain ID
+     * @return KlayRawTransaction this contains transaction hash and processed signature data
+     * @throws EmptyNonceException throw exception when nonce is null
+     */
+    @Override
+    public KlayRawTransaction sign(KlayCredentials credentials, int chainId) {
+        if (getNonce() == null) {
+            throw new EmptyNonceException();
+        }
+
+        KlaySignatureData senderSignatureData = getSignatureData(credentials, chainId);
+        addSenderSignatureData(senderSignatureData);
+        List<RlpType> rlpTypeList = new ArrayList<>(rlpValues());
+        List<RlpType> senderSignatureList = new ArrayList<>();
+
+        for (KlaySignatureData klaySignatureData : getSenderSignatureDataSet()) {
+            senderSignatureList.add(klaySignatureData.toRlpList());
+        }
+
+        rlpTypeList.add(new RlpList(senderSignatureList));
+        rlpTypeList.add(RlpString.create(Numeric.hexStringToByteArray(this.feePayer)));
+        if (this.feePayer.equals(EMPTY_FEE_PAYER_ADDRESS)) {
+            rlpTypeList.add(new RlpList(KlaySignatureData.createKlaySignatureDataFromChainId(1).toRlpList()));
+        } else {
+            List<RlpType> feePayerSignatureList = new ArrayList<>();
+            for (KlaySignatureData klaySignatureData : this.feePayerSignatureData) {
```

Comment:
If `feePayerSignatureData` have multiple SignatureData, whose signature pay for the fee?

Reply:
The fee payer is issued and all signatures must be signed by the fee payer.